File: AgoraChat-ObjectiveC/AgoraChat/Code/Custom/View/ACDImageTextButtonView.m
//
//  ACDImageTextButtonView.m
//  ChatDemo-UI3.0
//
//  Created by liang on 2021/10/21.
//  Copyright © 2021 easemob. All rights reserved.
//

#import "ACDImageTextButtonView.h"

@interface ACDImageTextButtonView ()

@end

@implementation ACDImageTextButtonView

- (instancetype)initWithFrame:(CGRect)frame
{
    self = [super initWithFrame:frame];
    if (self) {
        self.userInteractionEnabled = YES;
        self.translatesAutoresizingMaskIntoConstraints = NO;
        
        _iconImageView = UIImageView.new;
        _iconImageView.contentMode = UIViewContentModeScaleAspectFill;
        _iconImageView.userInteractionEnabled = NO;
        
        _titleLabel = UILabel.new;
        _titleLabel.textColor = TextLabelBlackColor;
        _titleLabel.textAlignment = NSTextAlignmentCenter;
        _titleLabel.font = NFont(12.0f);

        _tapBtn  = UIButton.new;
        
        [self addSubview:_tapBtn];
        [_tapBtn addSubview:_iconImageView];
        [_tapBtn addSubview:_titleLabel];

        [_tapBtn mas_makeConstraints:^(MASConstraintMaker *make) {
            make.edges.equalTo(self);
        }];
        
        [_iconImageView mas_makeConstraints:^(MASConstraintMaker *make) {
            make.top.equalTo(_tapBtn.mas_top).offset(5.0);
            make.centerX.equalTo(_tapBtn);
            make.size.mas_equalTo(40.0);
        }];

        [_titleLabel mas_makeConstraints:^(MASConstraintMaker *make) {
            make.top.equalTo(_iconImageView.mas_bottom).offset(5.0);
            make.centerX.equalTo(_tapBtn);
            make.height.equalTo(@12.0f);
            make.bottom.equalTo(_tapBtn.mas_bottom).offset(-5.0);
        }];
    }
    return self;
}

@end

File: AgoraChat-Swift/AgoraChat-Swift/code/Custom/View/ACDImageTextButtonView.swift
//
//  ACDImageTextButtonView.swift
//  AgoraChat-Swift
//
//  Created by 冯钊 on 2022/10/18.
//

import UIKit

class ACDImageTextButtonView: UIButton {
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        self.titleLabel?.textAlignment = .center
    }
    
    required init?(coder: NSCoder) {
        super.init(coder: coder)
        self.titleLabel?.textAlignment = .center
    }
    
    override func layoutSubviews() {
        super.layoutSubviews()
        
        self.imageView?.frame = CGRect(x: (self.bounds.width - 40) / 2, y: 5, width: 40, height: 40)
        self.titleLabel?.frame = CGRect(x: 0, y: 50, width: self.bounds.width, height: 12)
    }
}
